sway/desktop/render.c

```diff
@@ -364,7 +408,16 @@ static void render_surface_iterator(struct sway_output *output,
 	deco_data.corner_radius *= wlr_output->scale;
 
 	// render blur (view->surface == surface excludes blurring subsurfaces)
-	if (deco_data.blur && should_parameters_blur() && view->surface == surface) {
+	bool is_subsurface = true;
+	bool is_floating = true;
+	if (data->is_toplevel_surface) {
+		is_subsurface = false;
+		is_floating = !config->blur_xray;
```

Comment:
lets rename `is_floating` to `should_optimize_blur` or something more descriptive

Reply:
Fixed by: [7355f7c](https://github.com/WillPower3309/swayfx/pull/144/commits/7355f7c4e7951fc788b2891fb35087add3eafdc8)